src/main/java/com/artipie/asto/Copy.java

```diff
@@ -69,7 +62,7 @@ public Copy(final Storage from, final List<Key> keys) {
      * @return When copy operation completes
      */
     public CompletableFuture<Void> copy(final Storage to) {
-        return Single.error(new IllegalStateException("not implemented")).ignoreElement()
+        return new RxCopy(new RxStorageWrapper(this.from), this.keys).copy(new RxStorageWrapper(to))
```

Comment:
@Sammers21 If we will have a reactive version of copy, should really Copy rely on it? Will it be useful to have `Copy` (non reactive verison) in this case?

Reply:
Yes, `CompletableFuture` computational composition primitives are weak, that is why we should rely on the Rx version.